backend/generator/Generator.hs

```diff
@@ -0,0 +1,110 @@
+-- SPDX-FileCopyrightText: 2021 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: AGPL-3.0-or-later
+
+module Generator
+  ( GeneratorSpec (..)
+  , generateAndSave
+  ) where
+
+import Universum
+
+import qualified Data.ByteString.Lazy as BSL
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as Text
+
+import qualified Edna.Library.Service as Library
+import qualified Edna.Upload.Service as Upload
+
+import Edna.ExperimentReader.Types
+  (FileContents(..), FileMetadata(..), Measurement(..), TargetMeasurements(..))
+import Edna.Init (initEdna)
+import Edna.Library.Web.Types (MethodologyReq(..), ProjectReq(..))
+import Edna.Logging (logMessage)
+import Edna.Setup (Edna)
+import Edna.Util (SqlId(..))
+
+-- | Specification for data generator: determines what data will be generated.
+data GeneratorSpec = GeneratorSpec
+  { gsProjectNum :: Word32
+  -- ^ How many projects to generate.
+  , gsMethodologyNum :: Word32
+  -- ^ How many methodologies to generate.
+  , gsCompoundNum :: Word32
+  -- ^ How many compounds to put into one file.
+  , gsTargetNum :: Word32
+  -- ^ How many targets to put into one file.
+  , gsMeasurementsNum :: Word
+  -- ^ Number of measurements in one experiment.
+  }
+
+-- | Generate some dummy data and save it to DB according to the provided spec.
+generateAndSave :: GeneratorSpec -> Edna ()
+generateAndSave spec@GeneratorSpec {..} = do
+  initEdna
+
+  mapM_ addProject [1 .. gsProjectNum]
+  logMessage $ "Added " <> show gsProjectNum <> " projects"
+
+  mapM_ addMethodology [1 .. gsMethodologyNum]
+  logMessage $ "Added " <> show gsMethodologyNum <> " methodologies"
+
+  addExperiments spec
+  where
+    addProject i = Library.addProject ProjectReq
+      { prqName = "Project #" <> show i
+      , prqDescription = show i <$ guard (even i)
+      }
+
+    addMethodology i = Library.addMethodology MethodologyReq
+      { mrqName = "Methodology #" <> show i
+      , mrqDescription = show i <$ guard (even i)
+      , mrqConfluence = Nothing
+      }
+
+addExperiments :: GeneratorSpec -> Edna ()
+addExperiments spec@GeneratorSpec {..} = do
+  forM_ [1 .. gsProjectNum] $ \projectIdx -> do
+    forM_ [1 .. gsMethodologyNum] $ \methodologyIdx ->
+      addExperiment projectIdx methodologyIdx
+    logMessage $ "Processed project " <> show projectIdx
+  where
+    addExperiment :: Word32 -> Word32 -> Edna ()
```

Comment:
Maybe we should create type for `Word32`? for example `type Idx = Word32`. I see that we use same value as number of elements and as indexes.. but we use it as num only in `GeneratorSpec` and there we can leave definition as it is and use `Idx` only in other function's type signatures

Reply:
Okay, I'll define `type Idx = Word32` and use in this module. If you suggest defining it in the library, we may consider doing that, but it's a broader topic and out of scope for this PR.

Reply:
Done.

Reply:
> If you suggest defining it in the library, we may consider doing that, but it's a broader topic and out of scope for this PR.

I saw that we have some complex types in tests, and maybe in other places too. I think we should pay attention to it and define some types, we can create an issue to research it and make according changes